### PR TITLE
Add support for 2023.3

### PR DESCRIPTION
**Pull request by rli**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Description

#3913

## Checklist
- [x] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [x] A short description of the change has been added to the **[CHANGELOG](https://github.com/aws/aws-toolkit-jetbrains/blob/master/CONTRIBUTING.md#contributing-via-pull-requests)** if the change is customer-facing in the IDE.
- [ ] I have added metrics for my changes (if required)
 
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: jetbrains-gateway/src/software/aws/toolkits/jetbrains/gateway/CawsConnectionProvider.kt [hunk 1]
```diff
@@ -214,12 +214,12 @@ class CawsConnectionProvider : GatewayConnectionProvider {
                                     codecatalystDevEnvironmentWorkflowError = "Timeout/Unknown error while connecting to Dev Env via SSM",
                                     duration = timeTakenToCheckInstallation.toDouble()
                                 )
-                                coroutineScope {
-                                    launchChildIOBackground {
-                                        environmentActions.stopEnvironment()
-                                        GatewayUI.getInstance().connect(parameters)
-                                    }
+
+                                launchChildSyncIOBackground {
```
Comment (manodnyab) [inline jetbrains-gateway/src/software/aws/toolkits/jetbrains/gateway/CawsConnectionProvider.kt:218]: Is this compatible with 232 until we deprecate it?

**Review by manodnyab — COMMENTED** [review]
We also need a changelog and updating the readme

File: jetbrains-core/tst/software/aws/toolkits/jetbrains/core/explorer/AwsExplorerNodeProcessorTest.kt [hunk 2]
```diff
@@ -93,7 +93,7 @@ class AwsExplorerNodeProcessorTest {
             }
         }
 
-        countDownLatch.await(1, TimeUnit.SECONDS)
+        countDownLatch.await(10, TimeUnit.SECONDS)
```
Comment (manodnyab) [inline jetbrains-core/tst/software/aws/toolkits/jetbrains/core/explorer/AwsExplorerNodeProcessorTest.kt:96]: Woah, what step update required this bump?

Comment (rli) [reply jetbrains-core/tst/software/aws/toolkits/jetbrains/core/explorer/AwsExplorerNodeProcessorTest.kt:96]: not sure why the tests were timing out with 1s, but the test itself does not take that long

File: jetbrains-ultimate/resources/META-INF/IU.xml [hunk 3]
```diff
@@ -7,6 +7,4 @@
     <depends optional="true" config-file="ext-go.xml">org.jetbrains.plugins.go</depends>
     <depends optional="true" config-file="ext-ruby.xml">org.jetbrains.plugins.ruby</depends>
     <depends optional="true" config-file="ext-php.xml">com.jetbrains.php</depends>
-    <depends optional="true" config-file="ext-codewithme.xml">com.jetbrains.codeWithMe</depends>
```
Comment (manodnyab) [inline jetbrains-ultimate/resources/META-INF/IU.xml:10]: Why do we need to move these to a separate file?

**Review by manodnyab — APPROVED** [review]

File: jetbrains-ultimate/resources/META-INF/IU.xml (hunk 3 above)
Comment (rli) [reply jetbrains-ultimate/resources/META-INF/IU.xml:10]: tests try to load them and fail

**Pull request merged**